File: Dataset/Leetcode/valid/28/650.py
 class Solution(object):
    def XXX(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if haystack == needle:
            return 0
        
        if len(needle) == 0:
            return 0

        haystack_length = len(haystack)
        needle_length = len(needle)

        for i in range(haystack_length):
            # If the current char equals to 
            # needle[0]
            if haystack[i] == needle[0]:
                if haystack[i:i+needle_length] == needle:
                    return i
        
        return -1

File: Dataset/Leetcode/train/58/650.cpp
class Solution {
public:
    int XXX(string s) {
        int n = s.length();
        while(n > 0 && s[n-1] == ' ') n--;
        for(int i=n-1; i>=0; i--) {
            if(s[i] == ' ') return n-1-i;
        }
        return n;
    }
};

